azurecontainerapps/azurecontainerapps.ts

```diff
@@ -26,18 +26,27 @@ export class azurecontainerapps {
             // Set up array to store optional arguments for the 'az containerapp up' command
             const optionalCmdArgs: string[] = [];
 
-            // Get the path to the application source to build and run
+            // Get the path to the application source to build and run, if provided
             const appSourcePath: string = tl.getInput('appSourcePath', false);
 
-            // Get the previously built image to deploy
+            // Get the name of the ACR instance to push images to, if provided
+            const acrName: string = tl.getInput('acrName', false);
+
+            // Get the previously built image to deploy, if provided
             let imageToDeploy: string = tl.getInput('imageToDeploy', false);
 
-            // Ensure that either the application source or a previously built image is provided, but not both
-            if ((!appSourcePath && !imageToDeploy) || (!!appSourcePath && !!imageToDeploy)) {
+            // Ensure that appSourcePath and acrName are provided together
+            if ((!!appSourcePath && !acrName) || (!appSourcePath && !!acrName)) {
```

Comment:
Nit: feel free to ignore if you feel strongly, but the `!!` is often confusing for other developers not used to Javascript. Would using something like `appSourcePath != null` (that will also check for `undefined` as it's not a strict equal) be a potential equivalent? It obviously depends of the return value of the `getInput` method.

Reply:
(if we go this way, we could argue the same for other simple `!` that when applied to strings could be converted to `== null`, if we're lucky enough that `getInput` doesn't return weird values)

Reply:
That's a good call -- I added a new function in the `Utility.ts` class that will handle string validation moving forward:

```typescript
public isNullOrEmpty(str: string): boolean {
    return str === null || str === undefined || str === "";
}
```

I updated the if-statements in this file to use this new function to make it easier to read. Let me know if this sounds good!

Reply:
Yes that sounds good to me. Thanks!